### PR TITLE
More Bot Fixes

### LINKED ISSUE #13332: Player controlled ED209s can only fire their taser when targeting a mob.
They cant shoot in a direction by clicking a turf or object. they must be targeting a mob.

### LINKED ISSUE #13363: Safeties overload make bots unresponsive
When an AI overloads a bot's safeties, the bot becomes unresponsive, making it impossible to change its settings.

That's proper behavior if the safeties are removed by a crewmember with an emag, but it shouldn't happen when the AI does it.

### LINKED ISSUE #13377: Beepsky and securitrons don't retaliate if you beat them to destruction with your bare hands.
If you use the harm intent to beat beepsky or another securitron to death, they will never actually attempt to arrest you.

### LINKED ISSUE #13415: Clicking on a bot as cyborg from afar causes you to punch them if harm intent is on.
From ktccd|codern00b in #coderbus
>coderbus, I got literally seconds before I gotta leave for a lecture, so here goes: The new bots, at least medibot, can be hit from a distance by cyborgs! I tried clicking on it to change it's patrol, but I ended up hitting it since I had harm-intent. But I was in another room when I did it!

### PR DESCRIPTION
**Pull request by AnturK**
Fixes AI hacking. Fixes #13363
Fixes Beepsky ignoring punches. Fixes #13377
Makes ED-bots able to shoot at anything. Fixes #13332
Fixes #13415

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 1]
```diff
@@ -709,7 +709,6 @@ Pass a positive integer as an argument to override a bot's default speed.
 	//No ..() to prevent strip panel showing up - Todo: make that saner
 	if(topic_denied(usr))
 		usr << "<span class='warning'>[src]'s interface is not responding!</span>"
-		href_list = list()
```
Comment (GunHog) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: Do this test this? *Extensively?* Please ensure that anyone locked out using this proc cannot access the bot specific functions.

Comment (AnturK) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: Not sure what you mean, this is called regardless of href parameters.

Comment (GunHog) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: Example: Try to change the arrest settings on Beepsky (which is not covered in the Parent Topic) after you emag him.

File: code/modules/mob/living/simple_animal/bot/ed209bot.dm [hunk 2]
```diff
@@ -149,6 +150,15 @@ Auto Patrol[]"},
...
 			update_controls()
 
+/mob/living/simple_animal/bot/ed209/attack_hand(mob/living/carbon/human/M)
+	if(M.a_intent == "harm")
+		threatlevel = M.assess_threat(src)
+		threatlevel += 6
+		if(threatlevel >= 4)
+			target = M
+			mode = BOT_HUNT
+	return ..()
```
Comment (Aranclanos) [inline code/modules/mob/living/simple_animal/bot/ed209bot.dm]: you copypasted this

**Pull request merged**